pkg/sql/sem/tree/overload.go

```diff
@@ -134,7 +134,7 @@ type Overload struct {
 	SQLFn SQLFnOverload
 
 	// OnTypeCheck is incremented every time this overload is type checked.
-	OnTypeCheck func()
+	OnTypeCheck *func()
```

Comment:
I think this needs some explanation.. It's very infrequent that I see a function pointer...